app/controllers/api/v0/competitions_controller.rb

```diff
@@ -86,18 +86,15 @@ def podiums
   def event_results
     competition = competition_from_params(associations: [:rounds])
     event = Event.c_find!(params[:event_id])
-    results_by_round = competition.results
-                                  .where(event_id: event.id)
-                                  .group_by(&:round_type)
-                                  .sort_by { |round_type, _| -round_type.rank }
-    rounds = results_by_round.map do |round_type, results|
-      # I think all competitions now have round data, but let's be cautious
-      # and assume they may not.
-      # round data.
-      round = competition.find_round_for(event.id, round_type.id)
+    rounds = competition.results
```

Comment:
I actually also changed this code recently and I did
```
rounds_for_event = competition.rounds
                                  .filter { |round| round.event_id == event.id }
                                  .sort_by { |round| -round.round_type.rank }
rounds = rounds_for_event.map do |round|
  {
    id: round.id,
    roundTypeId: round.round_type.id,
    results: round.results.sort_by { |r| [r.pos, r.person_name] },
  }
```
Which does look much shorter, not sure if it is quicker though

Reply:
Went with the "middle ground" approach now :D